src/Perpetuum/Services/Weather/WeatherObservers.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Perpetuum.Reactive;
+using Perpetuum.Services.EventServices;
+using Perpetuum.Services.EventServices.EventMessages;
+using Perpetuum.Zones;
+using System;
+
+namespace Perpetuum.Services.Weather
+{
+    /// <summary>
+    /// Weather observer for transmitting EventMessages to EventListenerService
+    /// </summary>
+    public class WeatherEventListener : Observer<WeatherInfo>
+    {
+        private readonly IZone _zone;
+        private readonly EventListenerService _listener;
+        public WeatherEventListener(EventListenerService listener, IZone zone)
+        {
+            _zone = zone;
+            _listener = listener;
+        }
+
+        public override void OnNext(WeatherInfo info)
+        {
+            _listener.PublishMessage(new WeatherEventMessage(info, _zone.Id));
```

Comment:
This is because the current code uses an event listener service that responds to messages by type and then distributes them out, right? So this is basically our adapter between the event listener pattern and the RX pattern?

Reply:
Yes, most importantly the eventListener has a concurrentqueue of messages and processes them on a separate thread.  So this might appear redundant, but its adapting things for another event bus to get things off of what might be realtime sensitive threads or call stacks.